packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -117,77 +116,94 @@ export const App = ({
   const [isSignedIn, setIsSignedIn] = useState<boolean | undefined>(undefined);
 
   useEffect(() => {
-    if (!integration) {
-      return;
-    }
+    const initializeSession = async () => {
+      if (!integration) {
+        return;
+      }
 
-    // If the local apollo cache says we are signed in, then we can skip the rest.
-    if (isSignedInData?.isSignedIn === true) {
-      setIsSignedIn(true);
-      return;
-    }
+      // If the local apollo cache says we are signed in, then we can skip the rest.
+      if (isSignedInData?.isSignedIn === true) {
+        setIsSignedIn(true);
+        return;
+      }
+
+      // if there is already a valid current account, use it
+      const localUser = currentAccount();
+      if (
+        localUser?.sessionToken &&
+        (await session.isValid(localUser.sessionToken))
+      ) {
+        setIsSignedIn(true);
+        return;
+      }
 
-    (async () => {
       let isValidSession = false;
 
       // Request and update account data/state to match the browser state.
       // If there is a user actively signed into the browser,
       // we should try to use that user's account when possible.
-      const userFromBrowser = await firefox.requestSignedInUser(
-        integration.data.context,
-        // TODO with React pairing flow, update this if pairing flow
-        false,
-        integration.data.service
-      );
-
-      if (userFromBrowser && userFromBrowser.sessionToken) {
+      const ua = navigator.userAgent;
+      const isFirefox = /firefox/i.test(ua) || /FxiOS/i.test(ua);
```

Comment:
I don't think the match needs to be case insensitive.  And I think `string.includes` should work.